### DISCUSSION TITLE
B_GiveTradeInv

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/g2nr_unofficial_update#391] [Q&A]
....

**Comment by Kor-Angar** [comment]
B_GiveTradeInv_Gorax

два раза добавляется карта для Паладина. во 2ой и 3ой главах.

**Reply by dosinabox** [reply]
Как я понимаю, лишняя во второй главе? И когда добавляется в третьей главе? Не вижу в B_GiveTradeInv_Gorax.

**Reply by Kor-Angar** [reply]
странно. видимо я куда не в тот код посмотрел. сейчас смотрю у тебя один раз во второй главе создаётся.

Лучше всего создание перенести в диалог с Ультаром dia_ulthar_schreinevergiftet_info

кстати там где Ультар нахаляву даёт карту наверное забыли сделать проверку на то жив ли BRAHIM и есть ли у него карта.

То есть давать карту бесплатно наверное надо герою, если её нет в мире (герой убил Брахима и Горакса).

**Reply by dosinabox** [reply]
На случай мертвого Горакса и живого Ибрагима нет озвучки.

**Reply by dosinabox** [reply]
> Лучше всего создание перенести в диалог с Ультаром dia_ulthar_schreinevergiftet_info

Так и есть сейчас, ты наверное классику смотришь.

**Reply by Kor-Angar** [reply]
я не смотрел что там у Ультара так как  думал что ты дубль удалила у Горакса. 

Ввиду того что  ты уже перенесла, тогда можно просто удалить создание её у Горакса во второй главе  https://github.com/dosinabox/g2nr_unofficial_update/blob/e81e83248f85341c49d8386d58f7c821a77062ec/PrjGOTHIC/Story/B_GiveTradeInv/B_GiveTradeInv_Gorax.d#L53

**Reply by Kor-Angar** [reply]
```D
	if(!Npc_IsDead(Gorax))
	{
		CreateInvItems(Gorax,ItBE_Addon_NOV_01,1);
		CreateInvItems(Gorax,ItWr_Map_Shrine_MIS,1);
	};
```
я предлагал карту для послушника вставляться в продажу Ибрагиму ;-) если Гораксу вставлять это уже как то нечестно будет.

**Reply by dosinabox** [reply]
Мне кажется Гораксу ЛОРнее - послушник получает доступ к тайным знаниям, чтобы можно было молиться в дороге по разным поручениям. При этом не получая никакого преимущества. А паладину продают карту по служебным нуждам.

**Reply by Kor-Angar** [reply]
карта же является подсказкой для выполнения задания Пирокара. если ты добавишь карту в продажу Гораксу то по ней можно пещеру найти.

**Reply by dosinabox** [reply]
А если Ибрагиму при вступлении в послушники, а Гораксу при вступлении в маги?

**Reply by Kor-Angar** [reply]
можно. но тогда у Горакса лучше убрать обычную карту.

**Reply by dosinabox** [reply]
Мне не нравится идея давать Ибрагиму карту для послушника, потому что не вижу связи между картографом из городского порта и вступлением ГГ в монастырь. Пусть останется как есть. Горакса поменяю.
- Ибрагим: с началом второй главы для всех
- Горакс: при вступлении в маги и при начале квеста Ультара

**Reply by Kor-Angar** [reply]
тоже вариант

**Comment by Kor-Angar** [comment]
```      
 CreateInvItems      (slf,ItKe_Innos_MIS ,1);    //Joly: damit man wieder aus dem Kloster kommt, wenn man reingeklettert ist.
```   
проверку можно добавить на то что герой не маг или послушник

**Reply by dosinabox** [reply]
Ключ создается в инстанции Горакса и удаляется при вступлении в послушники.

**Reply by Kor-Angar** [reply]
а его в итоге выкупить то можно? :-)

**Reply by dosinabox** [reply]
можно

**Reply by dosinabox** [reply]
Создание ключа в инстанции Горакса избавляет от необходимости проверки гильдии в B_GiveTradeInv_Gorax().

**Reply by Kor-Angar** [reply]
ясно

**Reply by Kor-Angar** [reply]
```D
//		if((hero.guild != GIL_NOV) && (hero.guild != GIL_NONE))
//		{
//			CreateInvItems(slf,ItMi_RuneBlank,2);
//		};
```
а эти руны ты куда то перенесла? их можно создавать в момент принятия ГГ в маги огня.

**Reply by dosinabox** [reply]
Так и есть
```cpp
if(!Npc_IsDead(Gorax))
{
	CreateInvItems(Gorax,ItMi_RuneBlank,2);
};
if(!Npc_IsDead(Karras))
{
	CreateInvItems(Karras,ItMi_RuneBlank,1);
};
```

**Reply by Kor-Angar** [reply]
ну это же прекрасно. жаль что ты в коде комменты не пишешь :-)

**Reply by Kor-Angar** [reply]
пояс для мага кстати тоже лучше в диалоге Пирокара создавать вместе с рунами. а то послушник купит пояс Мага, а это нехорошо с точки зрения ЛОРа игры.

а пояс для послушника в момент приёма в послушники.

**Comment by dosinabox** [comment]
Итог по Гораксу:
- пояс послушника появляется в продаже только при вступлении в послушники
- пояс мага появляется в продаже только при вступлении в маги
- карта алтарей появляется в продаже с началом второй главы
- карта алтарей не создается повторно в диалоге с Ультаром
- карта алтарей доступна для всех гильдий, как у Ибрагима

**Reply by Kor-Angar** [reply]
> - карта алтарей появляется в продаже с началом второй главы
> - карта алтарей не создается повторно в диалоге с Ультаром

вроде наоборот: в продаже не появляется, а в диалоге создаётся :-)

**Reply by dosinabox** [reply]
Можно и так, но какой смысл скрывать эту карту от игрока, если такую же продает Ибрагим, на главу раньше и для всех?

**Reply by Kor-Angar** [reply]
эта карта вставляется на тот случай если у героя нет этой карты. там же у Ультара и проверка стоит про это, если карты у героя нет, то Ультар говорит что ты можешь  купить  карту у Ибрагима.

**Reply by dosinabox** [reply]
Если у ГГ есть карта, то Ультар ничего не говорит.
Если у ГГ нет карты, то Ультар говорит что её можно купить у Горакса.
Если Горакс мертв, то Ультар сам выдает карту ГГ бесплатно.
Про Ибрагима озвучки нет.

**Reply by dosinabox** [reply]
> Если у ГГ нет карты, то Ультар говорит что её можно купить у Горакса.

Что делать в этом случае? Создавать карту у Горакса в этот момент? Тогда она будет ограничена: только для паладинов и только в третьей главе. При том что у Ибрагима она есть со второй главы для всех.
Предлагаю создавать карту у Горакса со второй главы для всех, а в диалоге с Ультаром её проверять:
- у ГГ нет карты + у Горакса есть карта, значит ГГ её не купил -> сообщать что карту можно купить у Горакса
- у ГГ нет карты + у Горакса нет карты, значит ГГ её купил/отобрал и потерял/продал -> выдавать карту бесплатно
- у ГГ есть карта, значит ГГ её купил/отобрал у Горакса или Ибрагима -> ничего не делать

**Reply by dosinabox** [reply]
```cpp
AI_Output(self,other,"DIA_Ulthar_SCHREINEVERGIFTET_05_05");	//Горакс может продать тебе карту, на которой помечены наши алтари.
...
AI_Output(self,other,"DIA_Ulthar_SCHREINEVERGIFTET_05_06");	//Вот карта. На ней отмечены алтари, построенные нами.
```

**Reply by Kor-Angar** [reply]
> Что делать в этом случае? Создавать карту у Горакса в этот момент?

да.

> Тогда она будет ограничена: только для паладинов и только в третьей главе

такая авторская задумка

> При том что у Ибрагима она есть со второй главы для всех.

у него есть все карты, для всех.

> Предлагаю создавать карту у Горакса со второй главы для всех

во второй главе она никому не нужна. она вообще нужна послушнику в первой главе и паладину в третей главе.

***

вообще я не предлагал что то переделывать/менять, а просто заметил что для паладина она вставлялась два раза и про это написал :-) у меня даже есть теория что в ранних версиях герой становился паладином во второй главе, так как гора проверок на паладина во второй главе много.

***

если говорить о каких то переделках я бы предложил просто отдавать карту паладину без заморочек, но с диалогом в стиле ГГ

```
//Возьми эту святую воду и окропи ей основания алтаря.
//Со святыми словами очищения к алтарю вернется его былая сила.
// где мне их найти(искать)? 
//Горакс может продать тебе карту, на которой помечены наши алтари.
// но у меня нет денег
//Вот карта. На ней отмечены алтари, построенные нами.
//Теперь иди и выполняй свои поручения.
```

так же добавил бы в продажу карту Ибрагима  при вступления ГГ  в послушники.

**Reply by dosinabox** [reply]
> так же добавил бы в продажу карту Ибрагима при вступления ГГ в послушники.

Готово.
А для паладина добавлять в продажу только в диалоге с Ультаром, если жив Горакс?

> вообще я не предлагал что то переделывать/менять, а просто заметил что для паладина она вставлялась два раза и про это написал :-)

Как обычно берусь за что-то мелкое, а в итоге получается не мелкое)

**Reply by Kor-Angar** [reply]
> А для паладина добавлять в продажу только в диалоге с Ультаром, если жив Горакс?

да. если Горакс мёртв, то Ультар даёт карту,  иначе карта добавляется Гораксу.

**Comment by Kor-Angar** [comment]
еще есть предложение когда будет время перенести  примечание, для файлов B_GiveTradeInv  торговцев из Готики 2 МДК. 
В будущем будет проще :-)

**Comment by Kor-Angar** [comment]
в принципе и другие фракционные пояса можно добавлять в продажу при вступление в гильдию. ибо с какой резон продавать ментовской пояс не менту? это же почти что часть доспеха.

**Reply by Kor-Angar** [reply]
или нет? :-)

**Reply by dosinabox** [reply]
По-моему, пояс ополченца просто продается на рынке, поэтому его можно не ограничивать гильдией. Если бы в казармах или каким-либо ополченцем - то другое дело.
А кто продает пояс наемника?

**Reply by Kor-Angar** [reply]
по моему беннет продаёт пояс наёмника.
